creator-node/src/services/sync/skippedCIDsRetryService.js

```diff
@@ -0,0 +1,123 @@
+const Bull = require('bull')
+
+const models = require('../../models')
+const { logger } = require('../../logging')
+const utils = require('../../utils')
+const { serviceRegistry } = require('../../serviceRegistry')
+const { saveFileForMultihashToFS } = require('../../fileManager')
+const CIDFailureCountManager = require('./CIDFailureCountManager')
+
+const LogPrefix = '[SkippedCIDsRetryQueue]'
+
+/**
+ * TODO - consider moving queue/jobs off main process. Will require re-factoring of job processing / dependencies
+ */
+class SkippedCIDsRetryQueue {
+  constructor (nodeConfig, libs) {
+    if (!nodeConfig || !libs) {
+      throw new Error(`${LogPrefix} Cannot start without nodeConfig and libs`)
+    }
+
+    this.queue = new Bull(
+      'skipped-cids-retry-queue',
+      {
+        redis: {
+          port: nodeConfig.get('redisPort'),
+          host: nodeConfig.get('redisHost')
+        },
+        defaultJobOptions: {
+          // these required since completed/failed jobs data set can grow infinitely until memory exhaustion
+          removeOnComplete: true,
+          removeOnFail: true
+        }
+      }
+    )
+
+    // Clean up anything that might be still stuck in the queue on restart
+    this.queue.empty()
+
+    const SkippedCIDsRetryQueueJobIntervalMs = 5000 // nodeConfig.get('skippedCIDsRetryQueueJobIntervalMs')
+    const CIDMaxAgeMs = nodeConfig.get('skippedCIDRetryQueueMaxAgeHr') * 60 * 60 * 1000 // convert from Hr to Ms
+
+    this.queue.process(
+      async (job, done) => {
+        try {
+          await this.process(CIDMaxAgeMs, libs)
+        } catch (e) {
+          logger.error(`${LogPrefix} Failed to process job || Error: ${e.message}`)
+        }
+
+        // Re-enqueue job after some interval
+        await utils.timeout(SkippedCIDsRetryQueueJobIntervalMs, false)
+        await this.queue.add({ startTime: Date.now() })
+
+        done()
+      }
+    )
+  }
+
+  // Add first job to queue
+  async init () {
+    try {
+      await this.queue.add({ startTime: Date.now() })
+      logger.info(`${LogPrefix} Successfully initialized and enqueued initial job.`)
+    } catch (e) {
+      logger.error(`${LogPrefix} Failed to start`)
+    }
+  }
+
+  /**
+   * Attempt to re-fetch all previously skipped files
+   * Only process files with age <= maxAge
+   */
+  async process (CIDMaxAgeMs, libs) {
+    const startTimestampMs = Date.now()
+    const oldestFileCreatedAtDate = new Date(startTimestampMs - CIDMaxAgeMs)
+
+    // Only process files with createdAt >= oldest createdAt
+    const skippedFiles = await models.File.findAll({
```

Comment:
this query could get super large (unbounded). maybe we should enforce some limit (or pagination) here to be safe?

Reply:
i don't think this query could get that large, it will only look for files with skipped = true, and createdAt after date (configvar) - seems fine to me?

esp since the subsequent operations on returned files are sequential

Reply:
chatted offline -- randomness sounds like a nice easy win here

Reply:
done